src/viewer.js

```diff
@@ -978,6 +978,31 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
         this.forceRedraw();
     },
 
+    /**
+     * TODO
+     */
+    setAjaxHeaders: function(ajaxHeaders, propagate){
+        if (propagate === undefined) {
+            propagate = true;
+        }
+
+        if ($.isPlainObject(ajaxHeaders)) {
+            this.ajaxHeaders = ajaxHeaders;
+        }
+
+        if (propagate) {
+            for (var i = 0; i < this.world.getItemCount(); i++) {
+                this.world.getItemAt(i).setAjaxHeaders(null, propagate);
+            }
+
+            if (this.referenceStrip && this.referenceStrip.miniViewers) {
```

Comment:
Good catch! We should do the navigator as well.

Reply:
I didn't really understand if there is anything to do there, since it seems to use/override functions defined in the Viewer .

Reply:
Yeah, the navigator is basically just a little viewer, just like the thumbnails in the reference strip. I figure we just need:

```
if (this.navigator) {
  this.navigator.setAjaxHeaders(this.ajaxHeaders, true);
}
```

Reply:
Ok, I did as you suggested.